lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runime/azbatch/driver/AzureBatchResourceManager.java

```diff
@@ -69,32 +67,29 @@
   private final String azureBatchAccountKey;
   private final String jobId;
   private final AzureStorageUtil azureStorageUtil;
+  private final REEFEventHandlers reefEventHandlers;
   private final String taskId;
-  private final String AZ_BATCH_JOB_ID_ENV = "AZ_BATCH_JOB_ID";
+  private final String azBatchJobId = "AZ_BATCH_JOB_ID";
 
   @Inject
   AzureBatchResourceManager(
-      @Parameter(RuntimeParameters.ResourceAllocationHandler.class)
-      final EventHandler<ResourceAllocationEvent> resourceAllocationHandler,
-      @Parameter(RuntimeParameters.NodeDescriptorHandler.class)
-      final EventHandler<NodeDescriptorEvent> nodeDescriptorHandler,
       final LocalAddressProvider localAddressProvider,
       final REEFFileNames fileNames,
+      final REEFEventHandlers reefEventHandlers,
```

Comment:
What does this abstraction buy us except a simplified constructor?

Reply:
Not so much right now. But I imagine that this might grow in future. The benefit that I found was to have some similarity with the way it is structured in mesos and yarn.